sdk/keyvault/Azure.Security.KeyVault.Shared/src/ChallengeBasedAuthenticationPolicy.cs

```diff
@@ -225,27 +232,39 @@ private static AuthenticationChallenge ParseBearerChallengeHeaderValue(string ch
                         if (pair.Length == 2)
                         {
                             // We have a key and a value, now need to trim and decode
-                            string key = pair[0].Trim().Trim(new char[] { '\"' });
-                            string value = pair[1].Trim().Trim(new char[] { '\"' });
+                            string key = pair[0].AsSpan().Trim().Trim('\"').ToString();
+                            string value = pair[1].AsSpan().Trim().Trim('\"').ToString();
 
                             if (!string.IsNullOrEmpty(key))
                             {
-                                if (string.Equals(key, "scope", StringComparison.OrdinalIgnoreCase))
+                                // Ordered by current likelihood.
+                                if (string.Equals(key, "authorization", StringComparison.OrdinalIgnoreCase))
```

Comment:
Are we adding features?

Reply:
It actually wasn't correct before. I talked this over with @schaabs.

Reply:
Whould we add a test for it?

Reply:
It is being tested with the third test.